booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,654 @@
+      allDg((dg) -> dg.path("segregationGroups").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.DG_SEGREGATION_GROUPS)
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'inhalationZone' values must be from dataset",
+      allDg((dg) -> dg.path("inhalationZone").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.INHALATION_ZONE_CODE)
+    ),
+    JsonAttribute.allOrNoneArePresent(
+      JsonPointer.compile("/declaredValue"),
+      JsonPointer.compile("/declaredValueCurrency")
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The charges 'currencyCode' values must be from dataset",
+      mav -> mav.submitAllMatching("charges.*.currencyCode"),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.ISO_4217_CURRENCY_CODES)
+    ),
+    JsonAttribute.customValidator(
+      "The charges currency amount must not exceed more than 2 decimal points",
+      (body) -> {
+        var charges = body.path("charges");
+        var issues = new LinkedHashSet<String>();
+        for(JsonNode charge : charges) {
+          var currencyAmount = charge.path("currencyAmount").asDouble();
+          if (BigDecimal.valueOf(currencyAmount).scale() > 2) {
+            issues.add("Charge amount %s is expected to have 2 decimal precious ".formatted(currencyAmount));
+          }
+        }
+        return issues;
+      }
+    )
+    );
+
+  public static ActionCheck responseContentChecks(UUID matched, Supplier<CarrierScenarioParameters> cspSupplier, Supplier<DynamicScenarioParameters> dspSupplier, BookingState bookingStatus, BookingState amendedBookingState) {
+    var checks = new ArrayList<JsonContentCheck>();
+    checks.add(JsonAttribute.mustEqual(
+      CARRIER_BOOKING_REQUEST_REFERENCE,
+      () -> dspSupplier.get().carrierBookingRequestReference()
+    ));
+    checks.add(JsonAttribute.mustEqual(
+      BOOKING_STATUS,
+      bookingStatus.wireName()
+    ));
+    checks.add(ADVANCED_MANIFEST_FILING_CODES_UNIQUE);
+    checks.add(AMF_CC_MTC_COMBINATION_VALIDATIONS);
+    checks.add(SHIPMENT_CUTOFF_TIMES_UNIQUE);
+    checks.add(CHECK_CONFIRMED_BOOKING_FIELDS);
+    checks.add(VALIDATE_SHIPMENT_LOCATIONS);
+    checks.add(REQUESTED_CHANGES_PRESENCE);
+    checks.add(REASON_PRESENCE);
+    checks.addAll(STATIC_BOOKING_CHECKS);
+    if (CONFIRMED_BOOKING_STATES.contains(bookingStatus)) {
+      checks.add(COMMODITIES_SUBREFERENCE_UNIQUE);
+    }
+    if (BOOKING_STATES_WHERE_CBR_IS_OPTIONAL.contains(bookingStatus)) {
+      checks.add(JsonAttribute.mustBeAbsent(
+        CARRIER_BOOKING_REFERENCE
+      ));
+    }
+    generateScenarioRelatedChecks(checks, cspSupplier, dspSupplier);
+    return JsonAttribute.contentChecks(
+      BookingRole::isCarrier,
+      matched,
+      HttpMessageType.RESPONSE,
+      checks
+    );
+  }
+
+
+  private static JsonContentMatchedValidation combineAndValidateAgainstDataset(
```

Comment:
Rather than copy-wasting it into `BookingChecks` it would probably make sense to make a `public static` version of this in `JsonAttribute` and then remove the code duplication from the EblChecks.